core/src/main/java/fj/P1.java

```diff
@@ -237,10 +240,11 @@ public String toString() {
 		return Show.p1Show(Show.<A>anyShow()).showS(this);
 	}
 
+    @SuppressWarnings("unchecked")
     @Override
     public boolean equals(Object other) {
-        return !Equal.equalsValidationCheck(this, other) ? false :
-                Equal.p1Equal(Equal.<A>anyEqual()).eq(this, (P1<A>) other);
+        return Equal.equalsValidationCheck(this, other)
+                && Equal.p1Equal(Equal.<A>anyEqual()).eq(this, (P1<A>) other);
```

Comment:
sure @mperry . I thought that is too minor to have a separate PR. Affirmative to the point of tracking things separately.